### PR TITLE
feat: handle removed team member (AR-2477)

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Handling incremental events when the user is removed from a team 
- mark user as deleted in users db
- remove user from members list
- convert one on one conversation with this user to a group conversation
- remove user from above conversation members list

### Issues
One issue with this solution will be that on the conversation list we will not know anything about deleted user ( when we will have many conversations with deleted users it will be hard to find a proper one when we will want to find some conversation history with them). This works well for slow sync when we login on a fresh install (we don't have any conversations history).

In the next iteration, we should leave a conversation with a deleted user as one on one and mark it somehow that this user was deleted and ignore slow sync when it will try to update the conversation type and conversation members

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 1]
```diff
@@ -797,6 +800,17 @@ internal class ConversationDataSource internal constructor(
     override suspend fun updateConversationName(conversationId: ConversationId, conversationName: String, timestamp: String) =
         wrapStorageRequest { conversationDAO.updateConversationName(idMapper.toDaoModel(conversationId), conversationName, timestamp) }
 
+    override suspend fun deleteUserFromConversations(userId: UserId): Either<CoreFailure, Unit> =
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:803]: I think it is a better idea if we rewrite this to being performed in one Transactional query, so we access DB just one and succeed if all operations are good as well =)

**Review by yamilmedina — COMMENTED** [review]
Good direction, I think we can perform the query in just one shot =) 🚀

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt (hunk 1 above)
Comment (vitorhugods) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:803]: This too:
https://cashapp.github.io/sqldelight/multiplatform_sqlite/grouping_statements/

**Review by vitorhugods — APPROVED** [review]
I'd love to get test coverage for this.

But... ⏳

**Pull request merged**